src/sentry/migrations/0767_fix_old_group_first_seen_dates.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 5.1.1 on 2024-09-24 20:28
+
+from datetime import datetime, timezone
+
+from django.apps.registry import Apps
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBarApprox
+
+OLD_FIRST_SEEN_CUTOFF = datetime(2000, 1, 1, tzinfo=timezone.utc)
+
+
+def update_old_first_seen_dates(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    Group = apps.get_model("sentry", "Group")
+
+    for group in RangeQuerySetWrapperWithProgressBarApprox(
+        Group.objects.filter(first_seen__lt=OLD_FIRST_SEEN_CUTOFF),
+    ):
```

Comment:
I'd recommend removing `RangeQuerySetWrapperWithProgressBarApprox`. Using the range wrapper causes the data to be sorted by `id`, and since you're already filtering on `first_seen` it could cause the wrong index to be used.

There are only 35 rows here so this should be fine.

Reply:
Ah thanks for that tidbit - just doing `group in Group.objects.filter(...)` now